pkgs/racket-doc/scribblings/reference/eval-model.scrbl

```diff
@@ -540,46 +548,56 @@ For example, given the module declaration
 
 the evaluation of @racket[(require 'm)] creates the variable @racket[x]
 and installs @racket[10] as its value. This @racket[x] is unrelated to
-any top-level definition of @racket[x].
+any top-level definition of @racket[x]. It's as
+if the variable were actually named ``x in m.''
 
-@;- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+@;------------------------------------------------------------------------
 @subsection[#:tag "module-phase"]{Phases}
 
-A module can be @tech{instantiate}d in multiple @deftech{phases}. A
-phase is an integer that, again, is effectively a prefix on the names
-of module-level definitions. A top-level @racket[require]
+The purpose of @deftech{phases} is to solve the ``eval-when'' problem that
+appears in various Lisp dialects. In particular, phases
+address the necessary separation of names defined at execution time versus
+names defined at compile/macro-expansion time. See the paper
+``Composable and Compilable Macros'' for detailed information @cite["Flatt02"].
+
+A module can be @tech{instantiate}d in multiple @tech{phases}. A
+phase is an integer that, like a module name, is effectively a prefix on the names
+of module-level definitions. Phase 0 is the execution-time phase.
+
+A top-level @racket[require]
 @tech{instantiates} a module at @tech{phase} 0, if the module is not
 already @tech{instantiate}d at phase 0.  A top-level
-@racket[(require (for-syntax ....))] @tech{instantiates} a module at
+@racket[(require (for-syntax ...))] @tech{instantiates} a module at
 @tech{phase} 1 (if it is not already @tech{instantiate}d at that
-level); @racket[for-syntax] also has a different binding
+phase); @racket[for-syntax] also has a different binding
 effect on further program parsing, as described in
 @secref["intro-binding"].
 
-Within a module, some definitions are shifted by a phase already; the
+Within a module, some definitions are already shifted by a phase: the
 @racket[begin-for-syntax] form is similar to @racket[begin], but it
-shifts expressions and definitions by a relative @tech{phase} 1. 
+shifts expressions and definitions by a relative @tech{phase} +1. 
+Likewise, the @racket[define-for-syntax] form is similar to @racket[define],
+but shifts the definition by +1.
 Thus, if the module is @tech{instantiate}d at phase 1,
 the variables defined with @racket[begin-for-syntax] are created at phase 2,
 and so on. Moreover, this relative phase acts as another layer of
-prefixing, so that a @racket[define] of @racket[x] and a 
-@racket[begin-for-syntax]-wrapped
-@racket[define] of @racket[x] can co-exist in a module
+prefixing, so that @racket[x] defined with @racket[define] and
+@racket[x] defined with @racket[begin-for-syntax] can co-exist in a module
```

Comment:
Should this be “defined with `define-for-syntax`” or “defined with `define` wrapped with `begin-for-syntax`”?

Reply:
I'll make it `define-for-syntax`.